RecoParticleFlow/Benchmark/src/PFJetBenchmark.cc

```diff
@@ -313,42 +314,7 @@ void PFJetBenchmark::process(const reco::PFJetCollection& pfJets, const reco::Ge
 	  //	    << " has no track ref.." << std::endl;
 	  continue;
 	}
-	unsigned int iter = 0; 
-	switch (trackRef->algo()) {
-	case TrackBase::ctf:
-        case TrackBase::duplicateMerge:
-	case TrackBase::initialStep:
-	  iter = 0;
-	  break;
-	case TrackBase::lowPtTripletStep:
-	  iter = 1;
-	  break;
-	case TrackBase::pixelPairStep:
-	  iter = 2;
-	  break;
-	case TrackBase::detachedTripletStep:
-	  iter = 3;
-	  break;
-	case TrackBase::mixedTripletStep:
-	  iter = 4;
-	  break;
-	case TrackBase::pixelLessStep:
-	  iter = 5;
-	  break;
-	case TrackBase::tobTecStep:
-	  iter = 6;
-	  break;
-	case TrackBase::conversionStep:
-	  iter = 7;
-	  //std::cout << "Warning in entry " << entry_ << " : iter = " << trackRef->algo() << std::endl;
-	  //std::cout << ic << " " << *(constituents[ic]) << std::endl;
-	  break;
-	default:
-	  iter = 8;
-	  std::cout << "Warning in entry " << entry_ << " : iter = " << trackRef->algo() << std::endl;
-	  std::cout << ic << " " << *(constituents[ic]) << std::endl;
-	  break;
-	}
+	unsigned int iter = trackRef->algo()>6 ? 6: trackRef->algo();
```

Comment:
Also, please note that #8315 made an attempt to get rid of magic numbers and less/greater than comparisons of `TrackBase::algo()` (although https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc#L1409 got omitted and should be fixed; I hope in this PR if possible). I would prefer that the new code would not introduce new uses of either.

I have also a more practical question. The numerical value 6 corresponds to `pixelPairStep`. What should be done with phase1 `highPtTripletStep` (numerical value 22)? I would put it (as a reasonable first guess) along `initialStep` (4), but this code maps it to 6.